src/main/controllers/ChallengeController.kt

```diff
@@ -23,8 +24,17 @@ class ChallengeController: DefaultController<Challenge>(), RestController<Challe
         }
     }
 
+    @Throws(NotFoundException::class)
     override fun findOne(user: UserAccount, id: Int): SOAResult<Challenge> {
-        throw NotImplementedError()
+        val findChallengeResult = DaoService.execute {
+            Challenge.findById(id)!!
```

Comment:
so i would do this slightly diff --

the !! will throw an error if its not found -- so you wont be able to distinguish if this was a DB error or an error from the !!

suggest removing the !!

add a check after the SUCCESS check to check if the result data exists -- if it doesn't throw a Not found exception